packages/shared/src/components/squads/settings/SquadModerationSettingsSection.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement, useState } from 'react';
+import { SquadSettingsSection } from './SquadSettingsSection';
+import { Radio } from '../../fields/Radio';
+import { SourceMemberRole } from '../../../graphql/sources';
+import { Switch } from '../../fields/Switch';
+import { WidgetCard } from '../../widgets/WidgetCard';
+import { SimpleTooltip } from '../../tooltips';
+import { useToggle } from '../../../hooks/useToggle';
+
+interface SquadModerationSettingsSectionProps {
+  initialMemberPostingRole?: SourceMemberRole;
+  initialMemberInviteRole?: SourceMemberRole;
+  initialModerationRequired?: boolean;
+}
+
+const memberRoleOptions = [
+  {
+    label: 'All members (recommended)',
+    value: SourceMemberRole.Member,
+  },
+  {
+    label: 'Only Moderators',
+    value: SourceMemberRole.Moderator,
+  },
+];
+
+export function SquadModerationSettingsSection({
+  initialMemberInviteRole,
+  initialMemberPostingRole,
+  initialModerationRequired,
+}: SquadModerationSettingsSectionProps): ReactElement {
+  const [isHoveringSwitch, setIsHoveringSwitch] = useState(false);
+  const [memberPostingRole, setMemberPostingRole] = useState(
+    () => initialMemberPostingRole || SourceMemberRole.Moderator,
+  );
+  const [memberInviteRole, setMemberInviteRole] = useState(
+    () => initialMemberInviteRole || SourceMemberRole.Member,
+  );
+  const [moderationRequired, setModerationRequired] = useToggle(
+    initialModerationRequired,
+  );
+
+  return (
+    <WidgetCard heading="🔒 Moderation Settings">
+      <div className="flex flex-col gap-6 px-4 py-2">
+        <SquadSettingsSection
+          title="Post content"
+          description="Choose who is allowed to post new content in this Squad."
+          className="flex"
+        >
+          <Radio
+            name="memberPostingRole"
+            options={memberRoleOptions}
+            value={memberPostingRole}
+            onChange={(value) => setMemberPostingRole(value)}
+          />
+        </SquadSettingsSection>
+        <SquadSettingsSection
+          title="Require post approval"
+          description="Turn this on to have admins or moderators approve every new post before it's published."
+        >
+          <SimpleTooltip
+            placement="top-start"
+            content={
+              <span className="p- max-w-[188px] p-2 text-center">
```

Comment:
I don't think you need this whole span anyway? Just render as is.
The tooltip should handle the rendering for plain text ones.

Reply:
> I don't think you need this whole span anyway? Just render as is. The tooltip should handle the rendering for plain text ones.

You're right. I did it this way to make it look more like the design:
<img width="306" alt="Screenshot 2024-10-04 at 14 48 31" src="https://github.com/user-attachments/assets/211b0cac-b7a6-4e53-b6d7-e4eea19b78e1">

the design:
<img width="360" alt="Screenshot 2024-10-04 at 14 52 45" src="https://github.com/user-attachments/assets/522770db-b660-4adc-8c35-6480cc56703b">



Without the span it looks like this:
<img width="548" alt="Screenshot 2024-10-04 at 14 48 19" src="https://github.com/user-attachments/assets/1b6f8670-3d77-4b4f-a623-788e622a6adc">

Also on mobile the text is a little bit too long, so it creates an overflow and horizontal scroll.

Reply:
Hmm yeah probably something we should fix with our new context menu (if we get one)
To make it more inline maybe use this approach instead:
https://github.com/dailydotdev/apps/blob/ee7c37145463d0988b081608d3217e05e2fd2349/packages/shared/src/components/streak/popup/DayStreak.tsx#L79

Also maybe you can use the `show` variable to determine if it needs to render (as shown above in the example)

Reply:
> Hmm yeah probably something we should fix with our new context menu (if we get one) To make it more inline maybe use this approach instead:
> 
> https://github.com/dailydotdev/apps/blob/ee7c37145463d0988b081608d3217e05e2fd2349/packages/shared/src/components/streak/popup/DayStreak.tsx#L79
> 
> Also maybe you can use the `show` variable to determine if it needs to render (as shown above in the example)

@rebelchris Great tips!

I changed it to use the container instead of the span.

When it comes to the hover trigger: With your solution it allows us to remove the state, but the span needs to remain for the events

Great feedback!